### ISSUE TITLE
systemd user context issue

### ISSUE THREAD

**Issue by diggnsaegg** [containers/podman#8980]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

**Steps to reproduce the issue:**

1. Create new container as non-root user, run: `podman generate systemd --name --new rootless-nginx -f` , exec `systemctl --user daemon-reload`

2. start the service with `systemctl --user start container-rootless-nginx`

**Describe the results you received:**

podman systemd service fails with:

> Jan 14 22:14:38 localhost.localdomain systemd[1666]: container-rootless-nginx.service: New main PID 29453 does not belong to service, and PID file is not owned by root. Refusing.
Jan 14 22:14:38 localhost.localdomain systemd[1666]: container-rootless-nginx.service: New main PID 29453 does not belong to service, and PID file is not owned by root. Refusing.

**Describe the results you expected:**

A running nginx server running rootless

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
podman version 2.2.1
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.18.0
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.21-3.fc33.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.21, commit: 0f53fb68333bdead5fe4dc5175703e22cf9882ab'
  cpus: 12
  distribution:
    distribution: fedora
    version: "33"
  eventLogger: journald
  hostname: localhost.localdomain
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.10.6-200.fc33.x86_64
  linkmode: dynamic
  memFree: 24740642816
  memTotal: 33650270208
  ociRuntime:
    name: crun
    package: crun-0.16-3.fc33.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.16
      commit: eb0145e5ad4d8207e84a327248af76663d4e50dd
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    path: /run/user/1000/podman/podman.sock
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.8-1.fc33.x86_64
    version: |-
      slirp4netns version 1.1.8
      commit: d361001f495417b880f20329121e3aa431a8f90f
      libslirp: 4.3.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.0
  swapFree: 4294963200
  swapTotal: 4294963200
  uptime: 47m 31.49s
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/dan/.config/containers/storage.conf
  containerStore:
    number: 11
    paused: 0
    running: 0
    stopped: 11
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.3.0-1.fc33.x86_64
      Version: |-
        fusermount3 version: 3.9.3
        fuse-overlayfs: version 1.3
        FUSE library version 3.9.3
        using FUSE kernel interface version 7.31
  graphRoot: /home/dan/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 12
  runRoot: /run/user/1000/containers
  volumePath: /home/dan/.local/share/containers/storage/volumes
version:
  APIVersion: 2.1.0
  Built: 1607438270
  BuiltTime: Tue Dec  8 15:37:50 2020
  GitCommit: ""
  GoVersion: go1.15.5
  OsArch: linux/amd64
  Version: 2.2.1
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-2.2.1-1.fc33.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide?**

Yes

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Phsyical Fedora 33 machine, the same setup works fine with CentOS and another Fedora 33 machine I run

**Comment by Luap99** [comment]
Please paste the content of the service file. Did you create the container with `--userns keep-id` or a custom uid mapping?

**Comment by diggnsaegg** [comment]
I create the container with ansible (it is a demo for running a container rootless)

> - name: create nginx container
  containers.podman.podman_container:
    name: rootless-nginx
    image: docker.io/library/nginx
    image_strict: no
    state: present
    publish: 80:80
    volume:
            - "{{ podman_volume_path }}:/usr/share/nginx/html:Z"
  become_user: "{{ podman_user }}"

For testing purposes I tried to create the systemd unit files with the `--new` option and without.
Both with the same error.

Here are the files:

**created without --new**

> [Unit]
Description=Podman container-rootless-nginx.service
Documentation=man:podman-generate-systemd(1)
Wants=network.target
After=network-online.target
[Service]
Environment=PODMAN_SYSTEMD_UNIT=%n
Restart=on-failure
ExecStart=/usr/bin/podman start rootless-nginx
ExecStop=/usr/bin/podman stop -t 10 rootless-nginx
ExecStopPost=/usr/bin/podman stop -t 10 rootless-nginx
PIDFile=/run/user/1000/containers/overlay-containers/4664e43607d00a006f548efc440379ca5fcf74e3b2a0c74d4a1f8b9ddad665c0/userdata/conmon.pid
KillMode=none
Type=forking
[Install]
WantedBy=multi-user.target default.target

**created with --new option**

> [Unit]
Description=Podman container-rootless-nginx.service
Documentation=man:podman-generate-systemd(1)
Wants=network.target
After=network-online.target
[Service]
Environment=PODMAN_SYSTEMD_UNIT=%n
Restart=on-failure
ExecStartPre=/bin/rm -f %t/container-rootless-nginx.pid %t/container-rootless-nginx.ctr-id
ExecStart=/usr/bin/podman run --conmon-pidfile %t/container-rootless-nginx.pid --cidfile %t/container-rootless-nginx.ctr-id --cgroups=no-conmon -d --replace --name rootless-nginx --publish 80:80 --volume /home/dan/work/rootless-podman-demo/roles/rootless-podman-demo/files:/usr/share/nginx/html:Z --detach=True docker.io/library/nginx
ExecStop=/usr/bin/podman stop --ignore --cidfile %t/container-rootless-nginx.ctr-id -t 10
ExecStopPost=/usr/bin/podman rm --ignore -f --cidfile %t/container-rootless-nginx.ctr-id
PIDFile=%t/container-rootless-nginx.pid
KillMode=none
Type=forking
[Install]
WantedBy=multi-user.target default.target

**Comment by diggnsaegg** [comment]
I just set up another VM with Fedora 33 and Podman, well...setup is similar to my workstation, non-root user creating container, generate systemd unit file...works as expected - any suggestions where else I could take a look?

**Comment by Luap99** [comment]
Where did you put the service file?

**Comment by diggnsaegg** [comment]
Service file is in ~/.config/systemd/user

**Comment by Luap99** [comment]
@vrothberg @giuseppe Any ideas?

**Comment by giuseppe** [comment]
does it work better if you try `--cgroups=split` instead of `--cgroups=no-conmon` in the ExecStart directive?

**Comment by diggnsaegg** [comment]
Nope, this does nothing. The container itselfs is running fine rootless, only the managment via systemd does not work on my machine.

**Comment by diggnsaegg** [comment]
any further ideas where to take a look?

**Comment by vrothberg** [comment]
Are you starting the files via ansible?

**Comment by vrothberg** [comment]
I only saw this issue in https://access.redhat.com/solutions/4420581 or when setting the `User` attribute in a systemd unit running Podman.

**Comment by vrothberg** [comment]
I am going to close since I am convinced it's a systemd issue. But we can continue the conversation here.

**Comment by diggnsaegg** [comment]
> Are you starting the files via ansible?
I think you mean "generate" them? Yes, with the shell module and becoming the user who should run the service. But it also fails when I create them manually. Doesn't matter though :/

> I only saw this issue in https://access.redhat.com/solutions/4420581 or when setting the User attribute in a systemd unit running Podman (which is not supported).

I already found that which did not help in my case.

> I am going to close since I am convinced it's a systemd issue. But we can continue the conversation here.

Thank you for helping me :)

_________

As i mentioned earlier in this thread, I already tried a fresh install of Fedora 33 and that did work as expected. So I think my system is kinda messed up, but I am not sure where to look. I already tried to disable SELinux at a first step with no luck.

**Comment by vrothberg** [comment]
Can you try removing `/usr/share/containers/containers.conf` and re-run? Or did you run on a fresh system?

**Comment by diggnsaegg** [comment]
I removed the file and when I try to `systemctl --user start mycontainer.service` the commando hangs and then times out
But `systemctl --user status mycontainer.service` shows the following:

`Feb 09 21:19:16 localhost.localdomain systemd[1690]: Starting Podman container-httpd.service...
Feb 09 21:19:16 localhost.localdomain systemd[1690]: con-httpd.service: Found left-over process 15596 (podman pause) in control group while starting unit. Ignoring.
Feb 09 21:19:16 localhost.localdomain systemd[1690]: This usually indicates unclean termination of a previous run, or service implementation deficiencies.
Feb 09 21:19:16 localhost.localdomain systemd[1690]: con-httpd.service: New main PID 16300 does not belong to service, and PID file is not owned by root. Refusing.
Feb 09 21:19:16 localhost.localdomain systemd[1690]: con-httpd.service: New main PID 16300 does not belong to service, and PID file is not owned by root. Refusing.`

The container is running somehow:

`CONTAINER ID  IMAGE                           COMMAND           CREATED         STATUS             PORTS               NAMES
5b3923d7e9b1  docker.io/library/httpd:latest  httpd-foreground  45 seconds ago  Up 45 seconds ago  0.0.0.0:80->80/tcp  httpd`

But the systemd service is still failing
